### DISCUSSION TITLE
Merging a structure from two different data sources.

### DISCUSSION THREAD

**Discussion by rdirisio** [molstar/molstar#899] [Q&A] [answered]
Hello, thank you for the great repository; I find the plugin UI to be extremely convenient for development.

I consider this an extension of https://github.com/molstar/molstar/issues/397, where hadim was trying to merge structures from a PDB and many SDFs after they have been loaded. My question is similar, but I am struggling to implement it:

I would like to create a Structure to view PDB <--> SDF interactions, but  I cannot do this unless the two components are in one Structure. Currently, I am doing what is recommended in that issue:
```
plugin.builders.structure.rawData(...)
plugin.builders.structure.parseTrajectory(...)
plugin.builders.structure.createModel(...)
plugin.builders.structure.createStructure(...)
structures.push({
    ref: ...,
  });
```

for both the SDF and PDB data, and then I call the [`MergeStructures`](https://github.com/molstar/molstar/blob/25eb4450ad9feab81b54edb2485ca730cb8bee6b/src/apps/docking-viewer/index.ts#L185) state transform. 

The issue is that the original models/structures are still in the state tree. I would like to only show in the state tree the new merged structure. 

I see that @dsehnal mentioned in the issue 

> Or you might want to load the initial structures yourself, then you get the refs directly without having to query and delete anything. (see below)
For the SDF, you will have to find the associated ModelFromTrajectory function and call update on that or create multiple models at that point.

Ideally, I would have one model that combines the two structures from the two different data sources. Is that possible?

Thanks!

**Comment by dsehnal** [comment, marked as answer]
You can use `isGhost` to hide a node from the state tree: ``this.plugin.builders.data.rawData(..., { state: { isGhost: true } })`` if you don't want the nodes there.

It is also possible to do this with with just one subtree, but you would have to write your own custom transforms for that.

**Reply by rdirisio** [reply]
I see, I needed add `isGhost` to each component part along the way. I ended up doing it like this for pdb and sdf
```
  builder = plugin.build().toRoot();
  const pdbStructure = await builder.toRoot()
  .apply(StateTransforms.Data.RawData, { data: pdbData }, { state: { isGhost: true } })
  .apply(StateTransforms.Model.TrajectoryFromPDB, undefined, { state: { isGhost: true } })
  .apply(StateTransforms.Model.ModelFromTrajectory, undefined, { state: { isGhost: true } })
  .apply(StateTransforms.Model.StructureFromModel, undefined, { state: { isGhost: true } })
  .commit()
```
This looks a lot like the membrane example in the [cellpack extension](https://github.com/molstar/molstar/blob/25eb4450ad9feab81b54edb2485ca730cb8bee6b/src/extensions/cellpack/model.ts#L429-L438).

Thanks! It looks great.
